packages/core/src/internal/utils/dom.spec.ts

```diff
@@ -344,4 +345,14 @@ describe('Functional Helper: ', () => {
       expect(isVisible(element)).toBe(false);
     });
   });
+  describe('spanWrapper', () => {
+    it('wraps text nodes in an element', () => {
+      const element: HTMLElement = createTestElement();
+      const text = document.createTextNode('Hello spanWrapper');
+      element.appendChild(text);
+      spanWrapper(element.childNodes);
+      expect(element.children[0].tagName).toBe('SPAN');
+      expect(element.children[0].textContent).toBe('Hello spanWrapper');
+    });
```

Comment:
The text node wrapper will really help simplify a lot of our generic slots. Thanks for working on this!